### PR TITLE
Change/specific operations send receive socket

### LINKED ISSUE #402: [Change Request] remove backwards compatibility socketIO in v4
**Description**
There have been recent updates to the sockets as discussed in #167. Some things have been left intact for v3 (petronas) backwards compatibility, but they should be removed in v4.
This issue is a todo list of what these things are so we don't forget them.

- [x] Change rules for socket to have operation 'send' and 'receive' instead of 'view'/'create'. Or consider creating operations such as 'kill' for kill task. (this is not possible in v3 because all operations must be either view/create/delete/edit.
- [x] Remove the property 'complete' in returning task json. This option is superseded by the 'status' field that gives more detail.

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #402 

- Created new operations 'send' and 'receive' for socket rules
- Changed socket rules to reflect new operations
- Changes in endpoints to reflect changes in rules
- Removed the 'complete' property from db.Task which is superseded by 'status', but kept a function in its stead as that is used elsewhere
- Remove unused function

### CODE REVIEW

File: vantage6-server/vantage6/server/model/task.py [hunk 1]
```diff
@@ -37,9 +37,6 @@ class Task(Base):
     initiator = relationship("Organization", back_populates="created_tasks")
     init_user = relationship("User", back_populates="created_tasks")
 
-    # TODO remove this property in v4. It is superseded by status but now left
-    # here for backwards compatibility with other v3 versions
-    @hybrid_property
```
Comment (frankcorneliusmartin) [inline vantage6-server/vantage6/server/model/task.py:42]: Shouldn't the entire function `complete` be deleted?

File: vantage6-server/vantage6/server/resource/task.py [hunk 2]
```diff
@@ -488,7 +488,7 @@ def __verify_container_permissions(container, image, collaboration_id):
             return False
 
         # check master task is not completed yet
-        if db.Task.get(container["task_id"]).complete:
+        if db.Task.get(container["task_id"]).complete():
```
Comment (frankcorneliusmartin) [inline vantage6-server/vantage6/server/resource/task.py:491]: Why not use the `task.status` for this?

File: vantage6-server/vantage6/server/resource/token.py [hunk 3]
```diff
@@ -312,7 +312,7 @@ def post(self):
                 HTTPStatus.UNAUTHORIZED
 
         # validate that the task not has been finished yet
-        if db_task.complete:
+        if db_task.complete():
```
Comment (frankcorneliusmartin) [inline vantage6-server/vantage6/server/resource/token.py:315]: Same for this one

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]

File: vantage6-server/vantage6/server/model/task.py (hunk 1 above)
Comment (bartvanb) [reply vantage6-server/vantage6/server/model/task.py:42]: I have done a new commit that should address this comment and the comments below.
This property has now been fully removed, and in other places now making use of the `has_task_finished()` function from vantage6-common. Also, I removed the same property in the `result` class.

File: vantage6-server/vantage6/server/resource/task.py (hunk 2 above)
Comment (bartvanb) [reply vantage6-server/vantage6/server/resource/task.py]: See above

File: vantage6-server/vantage6/server/resource/token.py (hunk 3 above)
Comment (bartvanb) [reply vantage6-server/vantage6/server/resource/token.py]: See above

**Review by frankcorneliusmartin — APPROVED** [review]

**Pull request merged**